src/VizPOIAnalytics/index.js

```diff
@@ -0,0 +1,208 @@
+      };
+      this.setState({
+        hover: {
+          x,
+          y,
+          hoveredObject: object,
+          details,
+        },
+      });
+    } else {
+      this.setState({ hover: { x, y, hoveredObject: object } });
+    }
+  }
+
+  setUserLocation() {
+    const { viewport } = this.state;
+    navigator.geolocation.getCurrentPosition((position) => {
+      const newViewport = {
+        ...viewport,
+        latitude: position.coords.latitude,
+        longitude: position.coords.longitude,
+      };
+      this.setState({ viewport: newViewport });
+    });
+  }
+
+  getDataForCurrentViewport() {
+    const newBoundingBox = this.mapRef.getMap().getBounds();
+    const boundingBox = {
+      _ne: {
+        lng: newBoundingBox._ne.lng,
+        lat: newBoundingBox._ne.lat,
+      },
+      _sw: {
+        lng: newBoundingBox._sw.lng,
+        lat: newBoundingBox._sw.lat,
+      },
+    };
+    this.fetchAllPOIDetailsInCurrentViewport(boundingBox);
+  }
+
+  updateLayerSettings(settings) {
+    this.setState({ settings });
+  }
+
+  fetchAllPOIDetailsInCurrentViewport(boundingBox) {
+    fetch(
+      `https://map-api-direct.foam.space/poi/filtered?swLng=${boundingBox._sw.lng}&swLat=${boundingBox._sw.lat}&neLng=${boundingBox._ne.lng}&neLat=${boundingBox._ne.lat}&limit=10000&offset=0`,
+    )
+      .then((result) => result.json())
+      .then((arrayOfPOIObjects) => {
+        const points = arrayOfPOIObjects.map((item) => {
+          const temp = hexToDecimal(item.state.deposit);
+          const pointCoords = getPointCoords(item.geohash);
+          return {
+            position: [
+              parseFloat(pointCoords[0].toFixed(4)),
+              parseFloat(pointCoords[1].toFixed(4)),
+            ],
+            stakedvalue: temp,
+          };
+        });
+        this.setState({
+          points,
+        });
+      });
+  }
+
+  render() {
+    const {
+      hover, settings, points, viewport,
+    } = this.state;
+
+    if (!points.length) return null;
+
+    return (
+      <div>
+        {hover.details && (
+          <div
+            className="tooltipStyle"
+            style={{
+              transform: `translate(${hover.x}px, ${hover.y}px)`,
+            }}
+          >
+            <div className="">
+              <div>
+Latitude:
+                {' '}
+                {hover.details.latitude}
+              </div>
+              <div>
+Longitude:
+                {' '}
+                {hover.details.longitude}
+              </div>
+              <div>
+POI&apos;s:
+                {' '}
+                {hover.details.numOfPoints}
+              </div>
+              <div>
+                Accumulated sum of FOAM tokens:
+                {' '}
+                {hover.details.sumOfFoamTokens}
+              </div>
+              <div>
+                Accumulated value of FOAM tokens: $
+                {hover.details.sumValInUSD}
+              </div>
+            </div>
+          </div>
+        )}
```

Comment:
Formatted according to tooltip component from Deck.gl: https://github.com/uber/deck.gl/blob/master/docs/developer-guide/interactivity.md#using-react